tensorflow/java/src/main/java/org/tensorflow/Shape.java

```diff
@@ -98,4 +116,18 @@ public String toString() {
   }
 
   private long[] shape;
+
+  private boolean hasUnknownDimension() {
+    if (shape == null) {
+      return false;
```

Comment:
Let's make this `return true;`. After digging in a bit, I think that the fact that Python returns `True` for `TensorShape(None).__eq__(TensorShape(None))` is a bug.